src/devices/cpu/i386/i386ex.cpp

```diff
@@ -0,0 +1,316 @@
+#include "i386ex.h"
+#include "i386priv.h"
+
+DEFINE_DEVICE_TYPE(I386EX, i386ex_device, "i386ex", "Intel I386EX")
+
+i386ex_device::i386ex_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
+	: i386_device(mconfig, I386EX, tag, owner, clock, 16, 26, 16)
+	, m_386ex_io_config("io", ENDIANNESS_LITTLE, 16, 16, 0)
+	, m_uart0(*this, "uart0")
+	, m_uart1(*this, "uart1")
+	, m_pit(*this, "pit")
+	, m_pic_master(*this, "pic_master")
+	, m_pic_slave(*this, "pic_slave")
+{
+}
+
+void i386ex_device::device_reset()
+{
+	zero_state();
```

Comment:
Is it necessary to avoid calling  `i386_device::device_reset`?  It’s usually better to call the base `device_reset` and patch up things that need to be different.